src/lib/operators/projection.cpp

```diff
@@ -28,7 +28,23 @@ Projection::Projection(const std::shared_ptr<const AbstractOperator>& input_oper
                        const std::vector<std::shared_ptr<AbstractExpression>>& init_expressions)
     : AbstractReadOnlyOperator(OperatorType::Projection, input_operator, nullptr,
                                std::make_unique<OperatorPerformanceData<OperatorSteps>>()),
-      expressions(init_expressions) {}
+      expressions(init_expressions) {
+  /**
+   * Register as a consumer for all subplans of uncorrelated subqueries
+   */
+  for (const auto& expression : expressions) {
+    auto pqp_subquery_expressions = collect_pqp_subquery_expressions(expression);
+    for (auto& subquery_expression : pqp_subquery_expressions) {
+      // We do not register for the subplans of correlated subqueries because they are templated and cannot be
+      // executed without concrete parameters. Thus, there is no option for result sharing at this point.
```

Comment:
Mmmh, I can't follow the causality here. I would have expected for the comment to explain who registers as a consumer of those subqueries instead. If I understand your approach correctly, this is because uncorrelated subqueries are "owned" by the projection and reused across chunks while correlated subqueries cannot be reused by design and are immediately discarded in the EspressoEvaluator.